subsys/net/lib/CMakeLists.txt

```diff
@@ -3,6 +3,7 @@ add_subdirectory_if_kconfig(lwm2m)
 add_subdirectory_if_kconfig(sntp)
 add_subdirectory_ifdef(CONFIG_DNS_RESOLVER      dns)
 add_subdirectory_ifdef(CONFIG_MQTT_LIB          mqtt)
+add_subdirectory_ifdef(CONFIG_MQTT_SOCKET_LIB   mqtt_socket)
```

Comment:
Not sure about the MQTT_SOCKET prefix, what about calling it just MQTT and renaming the old mqtt as MQTT_LEGACY? The directory name we can change later when old MQTT library is removed but renaming config variable name is annoying later as people are using the old name.

Reply:
I didn't want to touch the existing implementation, until there was a positive feedback on the new one. Renaming it to MQTT_LEGACY sounds like a good idea though.